app/src/lib/git/fetch.ts

```diff
@@ -128,3 +129,37 @@ export async function fetchRefspec(
 
   await git(args, repository.path, 'fetchRefspec', options)
 }
+
+export async function fastForwardBranches(
+  repository: Repository,
+  branches: ReadonlyArray<IBranchBasicInfo>
+): Promise<void> {
+  const opts: IGitExecutionOptions = {
+    successExitCodes: new Set([0, 1]),
+    env: {
+      // This will make sure the reflog entries are correct after
+      // fast-forwarding the branches.
+      GIT_REFLOG_ACTION: 'pull',
+    },
+  }
+
+  const refPairs = branches.map(branch => `${branch.upstreamRef}:${branch.ref}`)
+
+  await git(
+    [
+      '-c',
+      'fetch.output=full',
+      'fetch',
+      '.',
+      '--show-forced-updates',
+      '-v',
+      // TODO: Once we upgrade to git 2.29 we can use --stdin to pass the refs,
+      // avoiding hitting any shell limitations related to the length of the
+      // command.
+      ...refPairs,
```

Comment:
I think we need to explicitly check that the `refPairs` (or `branches`) array is not empty before spreading here or else we could end up running `git fetch .` without any refs which would be redundant at best and will, if nothing else, clear `FETCH_HEAD`

Reply:
😱 😱 😱 😱 😱 😱